presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
         org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, handle.getSchemaTableName());
         return TableStatisticsMaker.getTableStatistics(typeManager, session, constraint, handle, icebergTable);
     }
+
+    @Override
+    public void createMaterializedView(ConnectorSession session, SchemaTableName viewName, ConnectorMaterializedViewDefinition definition, boolean replace, boolean ignoreExisting)
+    {
+        Map<String, String> properties;
+
+        Map buildProps = new HashMap();
+        buildProps.put(PRESTO_QUERY_ID_NAME, session.getQueryId());
+
+        // Generate a storage table name
+        String storageTableName = "st" + UUID.randomUUID().toString().replace("-", "");
+        buildProps.put(STORAGE_TABLE, storageTableName);
+        buildProps.put(PRESTO_VIEW_FLAG, "true");
+        if (replace && definition.getComment().isPresent()) {
+            buildProps.put(TABLE_COMMENT, definition.getComment().get());
+        }
+        else {
+            // Add a default table comment when the view is being created
+            buildProps.put(TABLE_COMMENT, "Presto Materialized View");
+        }
+        properties = ImmutableMap.<String, String>builder()
+            .putAll(buildProps)
+            .build();
+
+        io.prestosql.plugin.hive.metastore.Column dummyColumn = new io.prestosql.plugin.hive.metastore.Column("dummy", HIVE_STRING, Optional.empty());
+
+        String schemaName = viewName.getSchemaName();
+        String tableName = viewName.getTableName();
+        Table.Builder tableBuilder = Table.builder()
+                .setDatabaseName(schemaName)
+                .setTableName(tableName)
+                .setOwner(session.getUser())
+                .setTableType(org.apache.hadoop.hive.metastore.TableType.VIRTUAL_VIEW.name())
+                .setDataColumns(ImmutableList.of(dummyColumn))
+                .setPartitionColumns(ImmutableList.of())
+                .setParameters(properties);
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(VIEW_STORAGE_FORMAT)
+                .setLocation("");
+
+        HiveIdentity identity = new HiveIdentity(session);
+        Optional<Table> existing = metastore.getTable(identity, viewName.getSchemaName(), viewName.getTableName());
+
+        // It's a create command where the materialized view already exists and 'if not exists' clause is specified
+        if (!replace && existing.isPresent() && !ignoreExisting) {
+            throw new TableAlreadyExistsException(viewName);
+        }
+
+        List<ColumnMetadata> columns = new ArrayList<>();
+        for (ConnectorMaterializedViewDefinition.Column col : definition.getColumns()) {
+            columns.add(new ColumnMetadata(col.getName(), typeManager.getType(col.getType())));
+        }
+
+        Map<String, Object> finalProperties = new HashMap<>();
+        finalProperties.put(FILE_FORMAT_PROPERTY, FileFormat.PARQUET);
```

Comment:
This should use the file format from the table properties
```java
getFileFormat(definition.getProperties())

Reply:
Why modify the table properties here at all, rather than just passing them through?

Reply:
There are two distinct parts to create materialized view: creation of storage table and creation of view that points to the storage table. I have now restructured this function to show the two parts. Each object that gets created has its own 'table' properties e.g. view has presto_view flag and storage_table name. Storage table does not have those properties. I have now renamed the properties to indicate which object they correspond to.

The storage table properties are obtained from definition properties, if file format is not set, it takes the default as follows: 
```
Map<String, Object> storageTableProperties = new HashMap<>(definition.getProperties());
        if (!storageTableProperties.containsKey(FILE_FORMAT_PROPERTY)) {
            storageTableProperties.put(FILE_FORMAT_PROPERTY, DEFAULT_FILE_FORMAT_DEFAULT);
        }
```